api/controllers/builds.go

```diff
@@ -153,6 +158,104 @@ func BuildDelete(rw http.ResponseWriter, r *http.Request) *httperr.Error {
 	return RenderJson(rw, build)
 }
 
+// BuildImport imports a build for an app
+func BuildImport(rw http.ResponseWriter, r *http.Request) *httperr.Error {
```

Comment:
Almost all of this controller should be in the BuildImport provider method. The controller should have no knowledge of Docker, just handing the file upload straight off to `provider.BuildImport(app, reader)`